packages/common/src/store/purchase-content/sagas.ts

```diff
@@ -772,6 +809,257 @@ function* doStartPurchaseContentFlow({
+  return yield* call(signAndSendTransaction, transaction)
+}
+
+// We use our own version of getOrCreateAssociatedTokenAccount instead of the one
+// from @solana/spl-token so we can sign the transaction with the solana wallet provider
+// since we do not have access to the private key in the wallet provider.
+function* getUsdcAssociatedTokenAccountWithCreateInstruction({
+  connection,
+  sourceWalletPublicKey
+}: {
+  connection: Connection
+  sourceWalletPublicKey: PublicKey
+}) {
+  let instruction: TransactionInstruction | null = null
+
+  const destinationTokenAccountPublicKey = getAssociatedTokenAddressSync(
+    new PublicKey(TOKEN_LISTING_MAP.USDC.address),
+    sourceWalletPublicKey
+  )
+
+  try {
+    yield* call(getAccount, connection, destinationTokenAccountPublicKey)
+  } catch {
+    instruction = createAssociatedTokenAccountInstruction(
+      sourceWalletPublicKey,
+      destinationTokenAccountPublicKey,
+      sourceWalletPublicKey,
+      new PublicKey(TOKEN_LISTING_MAP.USDC.address)
+    )
+  }
+
+  return { destinationTokenAccountPublicKey, instruction }
+}
+
+function* purchaseWithAnything({
+  purchaserUserId,
+  contentId,
+  contentType = PurchaseableContentType.TRACK,
+  price,
+  extraAmount,
+  totalAmountWithDecimals,
+  inputMint
+}: {
+  purchaserUserId: ID
+  contentId: ID
+  contentType: PurchaseableContentType
+  price: number
+  extraAmount?: number
+  totalAmountWithDecimals: number
+  inputMint: string
+}) {
+  try {
+    const audiusSdk = yield* getContext('audiusSdk')
+    const sdk = yield* call(audiusSdk)
+    const audiusBackendInstance = yield* getContext('audiusBackendInstance')
+    const connection = yield* call(getSolanaConnection, audiusBackendInstance)
+
+    // Get the USDC user bank
+    const usdcUserBank = yield* call(getUSDCUserBank)
+    const usdcUserBankTokenAccount = yield* call(
+      getTokenAccountInfo,
+      audiusBackendInstance,
+      {
+        mint: 'usdc',
+        tokenAccount: usdcUserBank
+      }
+    )
+    if (!usdcUserBankTokenAccount) {
+      throw new Error('Failed to fetch USDC user bank token account info')
+    }
+
+    // Get the solana wallet provider
+    const provider = window.solana
+    if (!provider) return
+    const sourceWallet = yield* call(provider.connect)
+
+    // Swap input mint to usdc
+    console.info('Swapping to USDC then sending to USDC userbank...')
+    yield* call(swapToUsdcAndSendToUserbank, {
+      connection,
+      inputMint,
+      amount: totalAmountWithDecimals,
+      sourceWalletPublicKey: sourceWallet.publicKey,
+      usdcUserBankTokenAccountPublicKey: usdcUserBankTokenAccount.address,
+      signAndSendTransaction: provider.signAndSendTransaction
+    })
+
+    // Purchase from USDC user bank balance
+    console.info(
+      `Purchasing ${
+        contentType === PurchaseableContentType.TRACK ? 'track' : 'album'
+      } with id ${contentId}...`
+    )
+    if (contentType === PurchaseableContentType.TRACK) {
```

Comment:
consider not going through user bank?

Reply:
yeah, talked about this at the monetization sync yday, but we could probably do a "direct to recipient" flow here.
want to check this version in first just to make sure we hit timelines, but can def re-eval